### PR TITLE
Add new yast2_nfs_server test

### PR DESCRIPTION
**Pull request by Vogtinator**
This tests "yast2 nfs-server" by creating an NFS share,
writing a file to it and validating that the file is accessible after mounting.
It can also be used as a server in an "/ on NFS" test scenario.
In this case, NFSSERVER has to be 1, the server is accessible as
10.0.2.1 and it provides a mutex "nfs_ready".

Needles are provided by PR #58 on os-autoinst-needles-opensuse.

### CODE REVIEW

File: main.pm [hunk 1]
```diff
@@ -503,6 +503,7 @@ sub load_consoletests() {
         }
         loadtest "console/mtab.pm";
         if (!get_var("NOINSTALL") && !get_var("LIVETEST") && (check_var("DESKTOP", "textmode"))) {
+            loadtest "console/yast2_nfs_server.pm";
```
Comment (coolo) [inline main.pm]: I don't think we can test this generally.

Comment (coolo) [inline main.pm]: generally as in every textmode installation

Comment (Vogtinator) [inline main.pm]: It's quicker than the other two tests below (mysql and HTTP) and has similar scope/importance IMO. There's also only one test with DESKTOP=textmode AFAICS.

**Comment by Vogtinator** [comment]
The test uses match_has_tag now.

File: tests/console/yast2_nfs_server.pm [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    send_key 'delete';
+    send_key 'delete';
+    send_key 'delete';
+    type_string "rw,fsid=0,no_";
+    send_key 'alt-o';
+
+    # Done
+    assert_screen 'nfs-share-saved';
+    send_key 'alt-f';
+    wait_idle;
```
Comment (okurz) [inline tests/console/yast2_nfs_server.pm]: @coolo would you like this removed?

Comment (Vogtinator) [inline tests/console/yast2_nfs_server.pm]: Which lines are you referring to? GitHub shows me an entire section, which is definitely necessary for the test to be successful.

Comment (okurz) [inline tests/console/yast2_nfs_server.pm]: only the "wait_idle;"

Comment (Vogtinator) [inline tests/console/yast2_nfs_server.pm]: AFAICS clear_console does 'type_string "clear\n"', so without waiting some keys may get lost as the ncurses UI gabs the keyboard.

Comment (coolo) [inline tests/console/yast2_nfs_server.pm]: then change line 34 to echo something to $serialdev after the yast modules is done - and wait_serial for it

Comment (Vogtinator) [inline tests/console/yast2_nfs_server.pm]: Wouldn't that introduce a race condition if the string or even part of it is output exactly between 'alt-f' and wait_serial?
Very unlikely to happen, but wait_idle is much cleaner in this regard.

**Comment by coolo** [comment]
wait_serial checks all code written since the last wait_serial. And wait_idle is sleep 19;

**Comment by Vogtinator** [comment]
wait_idle replaced by wait_serial now.

File: main.pm (hunk 1 above)
Comment (coolo) [inline main.pm]: yeah, but they don't require qemu IPs.

Comment (Vogtinator) [inline main.pm]: What's the best place for it, then?

Comment (coolo) [inline main.pm]: At least guard it with a check_var('BACKEND', 'qemu') here

**Comment by Vogtinator** [comment]
Using a check for QEMU now.

**Comment by okurz** [comment]
LGTM

**Comment by sysrich** [comment]
LGTM, but not merging yet because it needs a matching main.pm change in the sle-12 branch and if I merged it right before going home I'd forget to do it myself.. @Vogtinator can either beat me to it or I'll tackle both tomorrow

**Comment by okurz** [comment]
After #920 got merged an update in the corresponding sle-main.pm is easy. @Vogtinator @sysrich who can solve it faster?

**Comment by Vogtinator** [comment]
I can do it (and also add a "wait_for_children" after the mutex_create in the test), but I've got a meeting right now so it may take a moment.

**Comment by Vogtinator** [comment]
Rebased to fix conflicts, added wait_for_children and the call to sle/main.pm.

**Comment by coolo** [comment]
https://openqa.suse.de/tests/281380/modules/yast2_nfs_server/steps/7 needs urgent fixing - it breaks all staging prjs

**Comment by okurz** [comment]
on it

**Pull request merged**